ecosystem/sep-0040.md

```diff
@@ -0,0 +1,160 @@
+
+
+## Motivation
+A standard interface for NFTs allows ecosystem applications (such as wallets or marketplaces) to create, view, or otherwise work with any NFT on Stellar.
+
+
+## Abstract
+Non-fungible tokens (NFTs) can describe many types of assets, from digital collectibles to physical items. We should also define a standard issuance model to explain how NFTs can and should be issued on the network. In order to maintain compatibility across the Stellar ecosystem and beyond, we need a standard way to describe and define them that is agnostic of what the NFT actually describes.
+
+
+## Specification
+We begin with the [issuance model](#issuance-model) and then cover the NFT [metadata specification](#nft-metadata-specification).
+
+### Issuance Model
+Given an arbitrary digital asset, here's how issuance typically works:
+
+  1. An account is created specifically to represent the NFT in the Stellar ledger.
+  2. The account configures itself with metadata describing the NFT.
+  3. The account issues a new Stellar asset representing ownership over said NFT.
+  4. Any account holding that asset lays claim to the NFT.
+
+Ownership of the token directly conveys ownership of a unit of the NFT, however that may be defined. Other scenarios such as locking down the issuing account to avoid supply increases, ownership transfers via payments, etc. are built on top of this basic model and are thus extraneous to this SEP.
+
+Note also that this SEP does not address concerns around immutability. While there is an optional checksum component (addressed later), the account itself is free to modify the data at-will if it is not locked.
+
+### NFT Specification
+This specification is partially inspired by Ethereum's [EIP-721 specification](https://eips.ethereum.org/EIPS/eip-721#specification) on NFTs; however, it is generalized to arbitrary asset types (i.e. not necessarily images).
+
+In order to minimize the costs of on-chain storage, we have multiple levels of indirection of NFT metadata. The only thing the NFT account stores on-chain is a description of where to find the metadata.
+
+#### NFT Account Description
+A proper NFT account should specify its `homedomain`. It should also configure the asset in its domain's `stellar.toml` file (see [SEP-1](https://stellar.org/sep-1#currency-documentation)) if appropriate.
+
+There's only one required field on the account itself:
+
+| Field       | Purpose |
+| ----------- | ------- |
+| `type`      | Describes how the "first hop" of metadata is defined, standard values include `ipfshash` and `stellar` |
+
+The two possible values for the `type` field describe two standard ways to store the metadata, and the details will be in a data entry corresponding to this value. In other words, `type: ipfshash` means that you should look under the `ipfshash` key data entry which further defines how to find the metadata.
```

Comment:
What about regular non-ipfs urls? these are pretty common in the eth nft land.  This is important if people will ever try to build nft bridges between eth and stellar.

Reply:
While I hate the idea of supporting storing assets on a centralized web server, you're right that we should probably include that option here. In fact, both this and the `ipfshash` field could be combined under a single `uri` data entry, but that'd be incompatible with Litemint's current model. I'll add a possible `type: uri` value.